sdk/textanalytics/ai-text-analytics/src/analyzeBatchActionsResult.ts

```diff
@@ -47,3 +155,312 @@ export interface PagedAnalyzeBatchActionsResult
    */
   statistics?: TextDocumentBatchStatistics;
 }
+
+/**
+ * The type of different actions supported by the begin analyze batch actions operation.
+ * @internal
+ */
+type TextAnalyticsActionType =
+  | "RecognizeCategorizedEntities"
+  | "RecognizePiiEntities"
+  | "ExtractKeyPhrases";
+
+/**
+ * The type of an action error with the type of the action that erred and its
+ * index in the list of input actions.
+ * @internal
+ */
+interface TextAnalyticsActionError {
+  /**
+   * A code describing the kind of error produced
+   */
+  readonly code: ErrorCode;
+  /**
+   * A message from the service explaining the error
+   */
+  readonly message: string;
+  /**
+   * The type of the action that erred
+   */
+  readonly type: TextAnalyticsActionType;
+  /**
+   * The index of the action that erred in the list of input actions
+   */
+  readonly index: number;
+}
+
+/**
+ * Converts the service task name (in the JSON pointer in an action error) to an action type name.
+ * @param serviceActionType - The task type name the service uses.
+ * @returns the action type name that the package uses
+ * @internal
+ */
+function convertTaskTypeToActionType(taskType: string): TextAnalyticsActionType {
+  switch (taskType) {
+    case "entityRecognitionTasks": {
+      return "RecognizeCategorizedEntities";
+    }
+    case "entityRecognitionPiiTasks": {
+      return "RecognizePiiEntities";
+    }
+    case "keyPhraseExtractionTasks": {
+      return "ExtractKeyPhrases";
+    }
+    default: {
+      throw new Error(`unexpected action type from the service: ${taskType}`);
+    }
+  }
+}
+
+/**
+ * Converts a service action error to one with the JSON pointer converted to an action index.
+ * @param erredActions - the action error the service sent
+ * @returns an action error with an action type and index
+ * @internal
+ */
+export function parseActionError(erredActions: TextAnalyticsError): TextAnalyticsActionError {
+  if (erredActions.target) {
+    const regex = new RegExp(
+      /#\/tasks\/(entityRecognitionTasks|entityRecognitionPiiTasks|keyPhraseExtractionTasks)\/(\d+)/
+    );
+    const res = regex.exec(erredActions.target);
```

Comment:
nit: consider renaming this to `result` as `res` can mean `response` or `result`